File: src/Back/Back.Domain/Models/RecipeItem.cs
using Back.Domain.Abstractions;
using Back.Domain.ValueObjects;

namespace Back.Domain.Models
{
    public class RecipeItem : Entity<RecipeItemId>
    {
        internal RecipeItem(RecipeId recipeId, IngredientId ingredientId, decimal quantity)
        {
            Id = RecipeItemId.Of(Guid.NewGuid());
            RecipeId = recipeId;
            IngredientId = ingredientId;
            Quantity = quantity;
        }

        public RecipeId RecipeId { get; private set; } = default!;
        public IngredientId IngredientId { get; private set; } = default!;
        public decimal Quantity { get; private set; } = default!;
    }
}

File: src/backend-java/src/main/java/recepti_type1/backend_java/Models/RecipeItem.java
package recepti_type1.backend_java.Models;
import jakarta.persistence.*;

import java.util.UUID;

@Entity
@Table(name = "\"RecipeItems\"")
public class RecipeItem
{
    @Id
    private UUID Id;
    private UUID RecipeId;
    private UUID IngredientId;
    private Float Quantity;

    public RecipeItem() {}

    public RecipeItem(UUID recipeId, UUID ingredientId, Float quantity)
    {
        this.Id = UUID.randomUUID();
        this.RecipeId = recipeId;
        this.IngredientId = ingredientId;
        this.Quantity = quantity;
    }

        public UUID getId() {
        return Id;
    }

        public void setId(UUID id) {
        this.Id = id;
    }

        public UUID getRecipeId() {
        return RecipeId;
    }

        public void setRecipeId(UUID recipeId) {
        this.RecipeId = recipeId;
    }

        public UUID getIngredientId() {
        return IngredientId;
    }

        public void setIngredientId(UUID ingredientId) {
        this.IngredientId = ingredientId;
    }

        public Float getQuantity() {
        return Quantity;
    }

        public void setQuantity(Float quantity) {
        this.Quantity = quantity;
    }
}
